File: goal-keeper-backend/src/main/java/com/goalkeeper/api/dto/TeamSummaryDataDto.java
package com.goalkeeper.api.dto;

import com.fasterxml.jackson.annotation.JsonRootName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Map;


@Builder
@Data
@JsonRootName("TeamSummaryData")
@NoArgsConstructor
@AllArgsConstructor
public class TeamSummaryDataDto {

    private String clubTeamName;
    private List<LabelValueDto> summaryData;

}

File: goal-keeper/src/data/dto/TeamSummaryDataDto.ts
import { JsonObject, JsonProperty } from 'json2typescript';
import LabelValueDto from "@/data/dto/LabelValueDto";

@JsonObject('TeamSummaryData')
export default class TeamSummaryDataDto {

    @JsonProperty('clubTeamName', String)
    public clubTeamName: string | undefined = undefined;

    @JsonProperty('summaryData', [LabelValueDto])
    public summaryData: LabelValueDto[] | undefined = undefined;
}
